### PR TITLE
Move manage_raid resource to environment and use volume resource on it.

### PR DESCRIPTION
**Pull request by enrico-usai**
### Description of changes
 * Move `manage_raid` resource to environment cookbook
 * Use `volume` resource to manage raid
   * Move wait_for_block_dev to environment. It is used by both volume and raid resources.
  * Adapt ChefSpec accordingly
  * Mark `raid_vol_array` as mandatory for mount and unmount actions only

### Tests
* Move ebs_mount hooks to config folder
* Fix ebs device check. nvme1n1 was added there as a placeholder, it was wrong.
* Add Inspec tests for RAID resource
* Define hooks for mount and unmount raid tests
* Add lifecycle hooks for ebs_unmount test

Executed:
* `bash kitchen.ec2.sh environment-config test raid-mount-alinux2`
* `bash kitchen.ec2.sh environment-config test raid-unmount-alinux2`
* `bash kitchen.ec2.sh environment-config test raid-unexport-alinux2`
* `bash kitchen.ec2.sh environment-config test ebs-unmount-alinux2`

### CODE REVIEW

File: cookbooks/aws-parallelcluster-environment/resources/manage_raid/partial/_raid_configuration.rb [hunk 1]
```diff
@@ -12,38 +12,23 @@
 
 property :raid_shared_dir, String, required: true
 property :raid_type, [String, Integer], required: %i(mount)
-property :raid_vol_array, Array, required: true
+property :raid_vol_array, Array, required: %i(mount unmount)
 
 action :mount do
```
Comment (dreambeyondorange) [inline cookbooks/aws-parallelcluster-environment/resources/manage_raid/partial/_raid_configuration.rb:17]: do we also want the `return if on_docker?` check for the mount action?

File: cookbooks/aws-parallelcluster-environment/libraries/volume.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    end
+    Chef::Log.info("device #{path} is ready")
+  end
+end
+
+#
+# Rescan the PCI bus to discover newly added volumes.
+#
+def rescan_pci
+  shell_out("echo 1 > /sys/bus/pci/rescan")
```
Comment (awslitvit) [inline cookbooks/aws-parallelcluster-environment/libraries/volume.rb:19]: Do we really need this separate function?

File: cookbooks/aws-parallelcluster-environment/resources/manage_raid/partial/_raid_configuration.rb [hunk 3]
```diff
@@ -12,38 +12,23 @@
...
   # Parse and determine RAID type (cast into integer)
   raid_type = new_resource.raid_type.dup.strip.to_i
-  # Parse volume info into an array
-  raid_vol_array.each_with_index do |vol, index|
-    raid_vol_array[index] = vol.strip
-  end
 
   # Attach each volume
   raid_dev_path = []
   raid_vol_array.each_with_index do |volumeid, index|
```
Comment (awslitvit) [inline cookbooks/aws-parallelcluster-environment/resources/manage_raid/partial/_raid_configuration.rb:25]: Nitpick: you don't need index, net of the volume name. This code could be further simplified.

File: cookbooks/aws-parallelcluster-environment/resources/manage_raid/partial/_raid_configuration.rb [hunk 4]
```diff
@@ -12,38 +12,23 @@
...
-  raid_shared_dir = "/#{raid_shared_dir}" unless raid_shared_dir.start_with?('/')
   # Parse and determine RAID type (cast into integer)
   raid_type = new_resource.raid_type.dup.strip.to_i
-  # Parse volume info into an array
-  raid_vol_array.each_with_index do |vol, index|
-    raid_vol_array[index] = vol.strip
-  end
 
   # Attach each volume
   raid_dev_path = []
```
Comment (awslitvit) [inline cookbooks/aws-parallelcluster-environment/resources/manage_raid/partial/_raid_configuration.rb:24]: Nitpick: no need to define an array.

File: cookbooks/aws-parallelcluster-environment/resources/manage_raid/partial/_raid_configuration.rb [hunk 5]
```diff
@@ -167,23 +119,19 @@
   # Detach each volume
   raid_vol_array.each_with_index do |volumeid, index|
     # Detach RAID EBS volume
-    execute "detach_raid_volume_#{index}" do
-      command "#{cookbook_virtualenv_path}/bin/python /usr/local/sbin/manageVolume.py --volume-id #{volumeid} --detach"
+    volume "detach raid volume #{index}" do
```
Comment (awslitvit) [inline cookbooks/aws-parallelcluster-environment/resources/manage_raid/partial/_raid_configuration.rb:122]: Nitpick: no need to use index.

File: cookbooks/aws-parallelcluster-environment/resources/manage_raid/partial/_raid_configuration.rb (hunk 1 above)
Comment (enrico-usai) [reply cookbooks/aws-parallelcluster-environment/resources/manage_raid/partial/_raid_configuration.rb:17]: yes, if you agree we can do it in another patch.

File: cookbooks/aws-parallelcluster-environment/resources/manage_raid/partial/_raid_configuration.rb [hunk 6]
```diff
@@ -167,23 +119,19 @@
...
+      volume_id volumeid
+      action :detach
     end
   end
 end
 
 action :unexport do
+  return if on_docker?
   raid_shared_dir = new_resource.raid_shared_dir.dup
```
Comment (awslitvit) [inline cookbooks/aws-parallelcluster-environment/resources/manage_raid/partial/_raid_configuration.rb:131]: Nitpick: no need to define this var.

File: cookbooks/aws-parallelcluster-environment/test/controls/ebs_spec.rb [hunk 7]
```diff
@@ -13,7 +13,7 @@
   only_if { !os_properties.on_docker? }
   describe mount('/shared_dir') do
     it { should be_mounted }
-    its('device') { should eq '/dev/nvme1n1' }
+    its('device') { should eq '/dev/xvdb' }
```
Comment (awslitvit) [inline cookbooks/aws-parallelcluster-environment/test/controls/ebs_spec.rb:16]: Is this essential? Do we use this device somewhere? If not, I'd remove it.

**Review by awslitvit — APPROVED** [review]

**Pull request merged**